### PR TITLE
Experimental #5002

### PR DESCRIPTION
**Pull request by sohje**
## Short Description:
Experimental wait at fort with lure module plugged
## Fixes/Resolves/Closes (please use correct syntax):
- #5002
- Bot will stay near fort till lure module expired
- Disabled by default 

To enable:
```js
      {
        "type": "MoveToFort",
        "config": {
            "enabled": true,
            "lure_attraction": true,
            "lure_max_distance": 1000,
            "wait_on_lure": true,
            "log_interval": 5
        }
      }
```

TODO:
- Need some tests (already spent 15 lure modules)
- Implement wait at fort for specified amount of time config option (like: wait_on_lure_max: "20", etc)
- Documentation update

### CODE REVIEW

**Comment by MZorzy** [comment]
question (can't undertand code ) when it stop near fort, gps noise is active or is extremely fixed?

**Comment by th3w4y** [comment]
@sohje I cannot see how you are waiting...

**Comment by sohje** [comment]
@th3w4y We filter forts by 'active_fort_modifier' if self.wait_at_fort enabled.
Always receiving fort with Lure module on get_nearest_fort() untill it expired.

File: pokemongo_bot/cell_workers/move_to_fort.py [hunk 1]
```diff
@@ -140,22 +150,26 @@ def _get_nearest_fort_on_lure_way(self, forts):
...
         forts = self.bot.get_forts(order_by_distance=True)
-
         # Remove stops that are still on timeout
-        forts = filter(lambda x: x["id"] not in self.bot.fort_timeouts, forts)
+        forts = filter(
+            lambda x: x["id"] not in self.bot.fort_timeouts or (
+                x.get('active_fort_modifier', False) and self.wait_at_fort
+            ),
+            forts
+        )
```
Comment (th3w4y) [inline pokemongo_bot/cell_workers/move_to_fort.py]: is this still removing the forts on timeout?

```
In [31]: wait_at_fort = True

In [32]: forts
Out[32]: 
[{'active': True, 'id': 10},
 {'active': True, 'id': 20},
 {'active': True, 'id': 40}]

In [33]: timeouts_ids
Out[33]: [10, 20, 30]

In [34]: filter(lambda q: q["id"] not in timeouts_ids or (q.get("active", False) and wait_at_fort), forts)
Out[34]: 
[{'active': True, 'id': 10},
 {'active': True, 'id': 20},
 {'active': True, 'id': 40}]

In [35]: 
```

Comment (sohje) [inline pokemongo_bot/cell_workers/move_to_fort.py]: Nope, cause we dont need to care about forts_on timeout here. We standing and waiting for lures and spin fort with lure module on timeout(check spin_fort.py update L186)

**Comment by z4ppy** [comment]
Nice PR ! thanks you :)

**Comment by solderzzc** [comment]
@sohje  Will it spin forts when wait on fort? is the `timeout` checking already in code?

**Comment by sohje** [comment]
@solderzzc Yeap, bot will spin forts.
```python
    def get_forts_in_range(self):
        forts = self.bot.get_forts(order_by_distance=True)
        now = time.time() * 1000
        forts = filter(lambda x: x.get('cooldown_complete_timestamp_ms', 0) < now, forts)

# forts = filter(lambda fort: fort["id"] not in self.bot.fort_timeouts, forts)
```

**Comment by solderzzc** [comment]
👍

**Comment by sohje** [comment]
@solderzzc re-approve and merge?

**Comment by solderzzc** [comment]
👍

**Pull request merged**